core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -61,6 +73,12 @@
   protected InetSocketAddress mAddress;
   protected TProtocol mProtocol;
 
+  /** Whether the client needs to handshake with master first. */
+  private static final boolean HANDSHAKE_NEEDED =
+      Configuration.getBoolean(PropertyKey.USER_CONF_CLUSTER_DEFAULT_ENABLED);
+  /** Whether the handshake is complete. */
+  private static AtomicBoolean sHandshakeComplete = new AtomicBoolean(false);
```

Comment:
Is a JVM only allowed to handshake once, even though there could be multiple clients?

Reply:
correct. this is because `Configuration` currently is a global state.